scripts/fast-reboot

```diff
@@ -305,10 +306,11 @@ function reboot_pre_check()
         exit ${EXIT_FILE_SYSTEM_FULL}
     fi
 
-    # Make sure that the next image exists
-    if [[ ! -d ${IMAGE_PATH} ]]; then
-        debug "Next image ${NEXT_SONIC_IMAGE} doesn't exist ..."
-        exit ${EXIT_NEXT_IMAGE_NOT_EXISTS}
+    # Verify the next image by sonic_installer
+    local message=$(sonic_installer verify_next_image 2>&1)
+    if [ $? -ne 0 ]; then
```

Comment:
This part of the reboot script is executed under -e option, meaning that the failure will trigger script to exit. Please take examples in the script to catch error code.

Reply:
Thanks, the command can be simplified, fixed.

Reply:
If you simplified this way, the `sonic_installer verify-next-image` will only output `Failed` or `Done`, which has no context in `fast-reboot` output.

As Ying suggested: Please take examples in the script to catch error code.

Your original `debug` command is still useful.

---
In reply to: [453135626](https://github.com/Azure/sonic-utilities/pull/979#discussion_r453135626) [](ancestors = 453135626)

Reply:
Thanks Qi, fixed.

Reply:
Thanks @yxieca actually